data/plugins/autocomplete.lua

```diff
@@ -640,9 +639,23 @@ command.add(predicate, {
     end
     if not inserted then
       local current_partial = get_partial_symbol()
-      doc:insert(line, col, text)
-      doc:remove(line, col, line, col - #current_partial)
-      doc:set_selection(line, col + #text - #current_partial)
+      local sz = #current_partial
+
+      for idx, line1, col1, line2, col2 in doc:get_selections(true) do
+        local n = col1 - 1
+        local line = doc.lines[line1]
+        for i = 1, sz + 1 do
+          local j = sz - i
+          local subline = line:sub(n - j, n)
+          local subpartial = current_partial:sub(i, -1)
+          if subpartial == subline then
+            doc:set_selections(idx, line1, col1, line2, n - j)
+            break
+          end
+        end
```

Comment:
I wonder if this should be simplified and only set the selection if the whole partial matches.
In what cases is the subpartial useful?

Also I wonder if it would be better to use `Doc:remove` instead of relying on selections getting overwritten (also because `undo` restores the selection...).